lib/rom/compat.rb

```diff
@@ -99,10 +100,23 @@ def relation_classes(gateway = nil)
       classes.select { |rel| rel.gateway == gw_name }
     end
 
+    # @api public
+    # @deprecated
+    def [](key)
+      gateways.fetch(key)
+    end
+
+    # @api public
+    # @deprecated
+    def gateways
+      @gateways ||= setup.components.gateways.map(&:build).map { |gw| [gw.config.name, gw] }.to_h
```

Comment:
`to_h` yields block starting 2.6 https://ruby-doc.org/core-2.6/Enumerable.html#method-i-to_h

Reply:
@flash-gordon hey sorry I missed this. I will open "a final PR" against "a virtual" branch once I'm done and happy with the outcome, so that you can go through diff and comment on things. Then I can address all-the-feedback. It's gonna be a massive diff of course but the code should be way more understandable than before.